controller/engine_controller.go

```diff
@@ -844,7 +844,6 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 
 	snapshots, err := engineClientProxy.SnapshotList(engine)
 	if err != nil {
-		engine.Status.Snapshots = map[string]*longhorn.SnapshotInfo{}
```

Comment:
In the past, we dissussed that we would deprecate `engine.Status.Snapshots`. However, I think we cannot deprecate `engine.Status.Snapshots` because it server as the source for a valid snapshot chain at a point in time. The list of snapshot CRs can not serve as a valid snapshot chain at a point in time. For example, the engines has 4 system snapshots. It tries to create 4 snapshot CRs but it failed once created 2 snapshot CRs. At this moment, the snapshot chains listed by snapshot CRs is invalid.